src/data-access/repositories/interfaces/device-repository.ts

```diff
@@ -1,5 +1,7 @@
 import { Device } from '../../../model/entities/device'
+import { INewsDevice } from '../../../model/entities/inews-device'
+import { TelemetricsDevice } from '../../../model/entities/telemetrics-device'
 
 export interface DeviceRepository {
-  getDevices(): Promise<Device[]>
+  findAllDevices(): Promise<(Device | INewsDevice | TelemetricsDevice)[]>
```

Comment:
To follow the naming convetion in our other Repositories, this should be called `getDevices(): Promise<Device>`